update-python-kata/README.md

```diff
@@ -0,0 +1,31 @@
+# FAQ: Update Python kata
+
+## What is the problem with old Python kata?
+
+The problem described here is caused by the fact that test suites of old kata do not conform to the form expected by Codewars code runner and produce test output in a format not conforming to the expected one. The main reason for that is that calls to assertions (`assert_equals`, `expect`, etc.) are not placed inside of an `@it` block, but in a scope of a `@describe` block or just in the top-level scope of tests.
+
+## Why was everything OK until now, and now the problem suddenly popped up?
```

Comment:
Just to be extra clear, I'm not planning to introduce breaking changes to the Python test framework anytime soon. So, it won't prevent adding new Python versions. I added the list to wiki because @Chrono79 had asked.

The main reason to update _for users_ is to improve the test output. Tests written with top-level assertions tend to lack structure and that results in poor output. Also, depending on how the test is written it can accidentally cause **deep** test result because the old format does not add `<COMPLETEDIN:>` automatically.

```text
test-1
  passed
  test-2
    passed
    test-3
      passed
      ... 100s of these
```

Making tests consistent is huge for maintainers. It'll also allow removing various hacks as you wrote from the runner like you wrote.
When updating, it should also follow the current best practices and import the solution explicitly because that's another hack done by the runner, and it can change the line numbers in errors.

---

I honestly don't like maintaining our own test framework (Qualified uses `unittest`), but Codewars currently assumes a single test framework per language for kata. I'd like to add support for more test frameworks in the future, but I haven't had the time to look into how much effort is required.

Updating to the new format should make future migration easier because it somewhat has a structure resembling tests. It'll also allow the test framework to be assertion-based.

Reply:
Main idea behind the FAQ is that some people after seeing the list rushed to work on it, and this in turn made some authors totally defensive and throwing "stfu it works" back. I want to explain to such people why the actions are needed, but maybe it's kata maintainers who should be held back?

Reply:
Ąbout improving test output: I think that getting rid of "test output entry per assertion" and emitting one green/red line per test case would make a huge difference here :)

Reply:
Yeah, I saw some comments by authors, and I'd like them to understand why as well.

I get that they're tired of updates or changes. But things change. I'm doing my best to minimize the effort required by them, but it's impossible to avoid them all.

Some authors translated to many languages themselves, so obviously they'll get notified about the necessary updates more often. They shouldn't have any reasons to complain especially if others are helping, and I'd like to know if there's any valid reasons to complain.

I don't know why, but some authors are just sensitive to the "Issue" label. "Suggestion" is enough to get access, so use that? "Issue" will be better when it must be updated as soon as possible.

Reply:
Maybe also clarify that the community will help when posting "Suggestion" comment and also thank them when it's resolved.

Reply:
> I think that getting rid of "test output entry per assertion" and emitting one green/red line per test case would make a huge difference here :)

That requires the test framework to be assertion-based which in turn requires all kata to use `@test.it`.

Reply:
You think it's ok now? would you like to have some pat modified in some particular way?

Reply:
Maybe it's better to have a general FAQ for maintenance and some guide for users who are helping to resolve the problem.

I basically want

- the authors/translator to understand the reasons to update their stuff in general, that they don't need to do them all by themselves, and how to cooperate (fix or at least not get in the way)
- the maintainers to understand how to communicate with the authors effectively

Reasons behind specific change can be documented separately. Maybe that can contain more info like severity or what can happen if it's not done (e.g., I might unpublish a beta kata that's not updated in a certain time frame, so we can move on).